### PR TITLE
[JENKINS-41437] Increase test coverage

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-42437](https://issues.jenkins-ci.org/browse/JENKINS-42437)

Increase coverage for:
- System scoped credentials.
- Personal scoped credentials.
- Credentials in custom domains.

@reviewbybees @stephenc @jglick

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 1]
```diff
@@ -38,74 +45,161 @@
...
+
+            // Look up all credentials
+            if (!globalDomain) {
+                DomainRequirement[] wrongIncludeDomainRequirements = {new HostnamePortRequirement("inclssude.com", 80)};
+                List<BaseStandardCredentials> credentialsWithWrongInclude = CredentialsProvider.lookupCredentials(BaseStandardCredentials.class, r.jenkins, auth, wrongIncludeDomainRequirements);
+                assertThat(credentialsWithWrongInclude.size(), is(0));
+
+                DomainRequirement[] excludeDomainRequirements = {new HostnamePortRequirement("exclude.com", 90)};
+                List<BaseStandardCredentials> credentialsWithExclude = CredentialsProvider.lookupCredentials(BaseStandardCredentials.class, r.jenkins, auth, excludeDomainRequirements);
+                assertThat(credentialsWithExclude.size(), is(0));
```
Comment (armfergom) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: Why not test that using the right domain requirements, you can find the credential?

Comment (varyvol) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: @armfergom I had actually tested them but they were not included in the PR. Thanks!

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 2]
```diff
@@ -35,77 +42,176 @@
...
+    public void secretFileSystemScopeTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.SYSTEM, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
+        testCreateUpdateDelete(credential, updatedCredential, null);
+    }
+
+    @Test
+    public void secretFilPersonalScopeTest() throws IOException, URISyntaxException {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: typo

Comment (varyvol) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: Fixed.

**Comment by varyvol** [comment]
@jglick yes, they are basic credential operations but with the credential types this plugin declares. Something incorrect in this implementation could potentially make things not to work correctly, and this tests increase the confidence on that not happening (like the two previously existing ones already did, but now with an increased scope).

**Comment by andresrc** [comment]
@stephenc ping?

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 3]
```diff
@@ -35,77 +42,176 @@
...
+    public void secretFileDomainTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
+
+        DomainSpecification[] ar = { new HostnamePortSpecification(INCLUDE_COMPLETE_HOST, EXCLUDE_COMPLETE_HOST)};
+        Domain d = new Domain("mydomain", "", Arrays.asList(ar));
+
+        testCreateUpdateDelete(credential, updatedCredential, d);
```
Comment (stephenc) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: I do not see the value in these tests. They just re-test core functionality in credentials plugin that cannot be broken by credentials impls. Adding useless tests is not a good idea imho

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java [hunk 4]
```diff
@@ -46,7 +47,7 @@
     @Test(expected = IllegalArgumentException.class)
     @Issue("JENKINS-30926")
     public void shouldThrowAnExceptionIfFileNameIsBlank() throws IOException {
-        new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "", new StubFileItem(), "", "");
+        new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "", new StubFileItem(), "", SecretBytes.fromString(""));
```
Comment (stephenc) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: This seems ok

**Comment by varyvol** [comment]
@stephenc @jglick fair enough. I have raised [pr#9](https://github.com/jenkinsci/plain-credentials-plugin/pull/9) just to remove the use of deprecated constructors in the tests.

**Comment by stephenc** [comment]
@varyvol you should verify my claim that the functionality you were testing is already tested in the credentials plugin... and if my claim is false then you could port some of this test code there

**Comment by varyvol** [comment]
@stephenc I did. There is coverage for global, system and user credentials, as well as domains with requirements. That's why I closed this.

**Comment by stephenc** [comment]
@varyvol just wanted to be sure we were not throwing away some test cases. They do not make sense here, but *if* there were some test cases missing against the credentials plugin *then* that is where we should have them... that and never trust one's own memory!

**Pull request closed without merging**